hydra-cluster/src/Hydra/Cluster/Faucet.hs

```diff
@@ -56,16 +56,38 @@ seedFromFaucet ::
   IO UTxO
 seedFromFaucet node@RunningNode{networkId, nodeSocket} receivingVerificationKey lovelace tracer = do
   (faucetVk, faucetSk) <- keysFor Faucet
-  retryOnExceptions tracer $ submitSeedTx faucetVk faucetSk
-  waitForPayment networkId nodeSocket lovelace receivingAddress
+  payments <- findPaymentsWithSameLovelace
+  case payments of
+    Just currentPayments -> do
+      signedTx <- retryOnExceptions tracer $ submitSeedTx faucetVk faucetSk
+      void $ awaitTransaction networkId nodeSocket signedTx
+      newPayments <-
+        waitForPayments networkId nodeSocket lovelace receivingAddress
+      pure
+        $ UTxO.fromPairs
+          . filter
+            ( \(txIn, _) ->
+                not $ txIn `member` UTxO.inputSet currentPayments
+            )
+        $ UTxO.pairs newPayments
+    Nothing -> do
+      void $ retryOnExceptions tracer $ submitSeedTx faucetVk faucetSk
+      waitForPayments networkId nodeSocket lovelace receivingAddress
```

Comment:
Should: This is VERY complicated

Why not just use `awaitTransaction` and then find `theOutput` in it?

```suggestion
   seedTx <- retryOnExceptions tracer $ submitSeedTx faucetVk faucetSk
   producedUTxO <- awaitTransaction networkId nodeSocket seedTx
   pure $ UTxO.filter (== theOutput) producedUTxO
```

Reply:
worked like a charm!

good catch!